rustls/src/client/hs.rs

```diff
@@ -511,8 +511,8 @@ impl State for ExpectServerHello {
 
         match server_version {
             TLSv1_3 if tls13_supported => {
-                sess.common.negotiated_version = Some(TLSv1_3);
-            }
+                sess.common.negotiated_version = TLSv1_3
```

Comment:
Here we should continue to set `sess.common.negotiated_version = Some(negotiated_version)` because this is what is reported by `get_protocol_version()`.

However, also from this point, we should keep passing the negotiated version through to the states that need it in each `into_expecting()` function simlar to https://github.com/ctz/rustls/pull/460.

Reply:
I refactored the setting of `sess.common.negotiated_version` as part of my latest changes, but it is still explicitly set.